src/query/interpreter.cpp

```diff
@@ -772,6 +865,59 @@ Callback HandleReplicationQuery(ReplicationQuery *repl_query, const Parameters &
                                   fmt::format("Replica {} is registered.", repl_query->replica_name_));
       return callback;
     }
+    case ReplicationQuery::Action::REGISTER_MAIN: {
+      if (!license::global_license_checker.IsEnterpriseValidFast()) {
+        throw QueryException("Trying to use enterprise feature without a valid license.");
+      }
+#ifdef MG_ENTERPRISE
```

Comment:
We usually do
```
#ifdef MG_ENTERPRISE
if (!license::global_license_checker.IsEnterpriseValidFast()) {
  throw QueryException("Trying to use enterprise feature without a valid license.");
}
```
Double check if the other way round is valid

Reply:
The reason I do this is because cypher_main_visitor has no notion of enterprise version of the code. So in the interpreter, for community version, I want to disable the possibility of calling this method.